traits/tests/test_expression.py

```diff
@@ -53,3 +64,74 @@ def _bar_default(self):
         msg = "The 'bar' trait of a Foo instance must be a valid"
         with self.assertRaisesRegex(TraitError, msg):
             f.bar
+
+    @unittest.expectedFailure  # FIXME issue #1096
```

Comment:
It would be better to wrap the part of the code you expect to fail using `assertRaises` instead of wrapping the entire test with `expectedFailure`. Despite the name, this decorator does not differentiate test errors from test failures. Test errors are considered expected failures as well and the test can end up being unmaintained.

Reply:
I'm fine with an expected failure if there's a plan to resolve this in the near future (e.g., before the 6.1 release).

Reply:
An alternative would be to remove these tests from this PR and attach them to the corresponding issue instead. Then the PR that fixes the issue can re-instate them.

Reply:
Wrapped the tests in `assertRaises` because the fix to the bug is not as straight forwards as expected.